### DISCUSSION TITLE
goto-definition opens buffer in new tab

### DISCUSSION THREAD

**Discussion by rallyram** [scalameta/nvim-metals#381] [Q&A] [answered]
I would be nice if one could configure opening a definition or implementation in a new tab instead of the current tab.

Is this possible currently?

Thanks!

**Comment by ckipp01** [comment, marked as answer]
Hey @rallyram thanks for the question! Technically, sure it's possible, but you'd probably have to hack it in yourself since I don't think there is a quick way to do it by default. The goto definition functionality is actually just part of Neovim, `nvim-metals` doesn't do anything specific with it, but forward it to Neovim. Check out `:h lsp-handler` and it'll give you an idea of how to override a handler (which is what the result of the definition is). You can see a short discussion about this [here](https://neovim.discourse.group/t/go-to-definition-in-new-tab/1552)
